### DISCUSSION TITLE
Live preview during capture and question about level shifter IC

### DISCUSSION THREAD

**Discussion by NWSpitfire** [gusmanb/logicanalyzer#156] [Q&A]
Hello!

I have recently discovered your project, and I must say I am extremely impressed. I have been playing with it for a little while and it performs very well. 

I just wondered if it is possible (or if I have missed it in the documentation). Is there a way to "preview" or see (some) of the data being captured, live as it is being captured? Or do a continuous capture? Is this possible?

Also, I am planning to design some custom PCB's to use with this (I would like to rackmount the pico's) but I was having trouble finding the 200mbps uni-directional level shifters (TXU0108 I think they were?). 

I just wondered if I could substitute them for SN74LVC541A or TXS0108E (although this is only 100mbps) without causing too many issues? I'm assuming you picked the original level shifters as they were rated for 200mbps? So by using either I would have to deal with a lower bandwidth.

Thank you for this excellent development, and for your help.

Kind regards,
NWSpitfire

**Comment by JimDrewGH** [comment]
I asked about the live (realtime) viewing of data and was told that the issue is the USB transfer speed is too slow (12Mbps max).   You would need a highspeed device (as opposed to the fullspeed device that the Picos are) for anything other than low KHz sampling.

The level shifters are TXU0104, 4 bit level shifters.  They don't make a 0108 in this particular family, which is better part (IMHO) than the TXS series.  I have had nothing but problems with bus voltage leakage with the TXS parts.

**Reply by NWSpitfire** [reply]
That makes sense, to be honest the more I use it the less I'm worried about it. But I did wonder so it's good to know, thank you!

That's right, forgive me I got mixed up with the TXS versions. That's interesting, so you have had 5v "high" voltage leak through to the "low" voltage side? Have you had issues with signal integrity?

If they are no good then I will probably try the SN74LVC541 buffers (I got these from other logic analyzers). I will have a look to see if I can reasonably source the TXU0104 at the same time though.

Thank you

**Comment by JimDrewGH** [comment]
Yes, what happens is that the 5V side raises the 3.3V side to about 3.7V when the associated pin is driven high.  That of course is a problem for anything that is not tolerant.   I did not experience any damage to the micros being used (3.6V tolerant PICs and ESP32), but it's not right.

I typically use something that has dedicated rail input voltages for level shifting, especially for bi-directional requirements.

**Reply by mengstr** [reply]
Have you measured the current for the 3.7v leakage? The RP2040 have  internal clamping diodes and should support virtually any voltage as long as the current is low enough. (Microchip app notes AN182 and AN521) On the other hand this will slow down the edges so it might not be a good solution for 200MHz operations ;-)

**Comment by JimDrewGH** [comment]
No, I have not measured the current.  I have 1K pull-ups on the 5V side with nothing on the 3.3V side other than the micro's (INPUT) pin.  Just sitting idle (even with the micro removed) the voltage is about 3.7V.  That is with the normal 200K pull-up reference.  If I change that reference to 470K I can get the voltage under 3.5V.  It could be that there is so little current that it really doesn't hurt anything, but again, it's not right.  I tried a couple of these TXS and TXE series with a couple of projects and always ended up going back to a part that has dedicated rails for each side (A/B).  The newer parts have sub-10ns propagation delay (some are down in the 2-3ns range).

**Reply by mengstr** [reply]
Now I'm getting a bit confused here - I thought you were speaking about the TXS0108E. But that device do have two power rails (even if the rails use have voltages where A is less than B).

**Reply by JimDrewGH** [reply]
Ack!  You are correct.  I was referring to a different family - the LSF chips.  These use voltage references for rails.  I could not use the TXS/TXE series because the propagation delays were too slow and they could not drive enough current for my application.  I ended up always using the SN74LVC8T245 (TI) or 74LVC8T245 (Nexperia).  These are basically identical.  The source/sink current is great (+/- 32mA) and the real-world propagation delays for 5V<>3.3V is <3ns.
